library/src/lib/alert/alert.component.ts

```diff
@@ -42,7 +51,45 @@ export class AlertComponent implements OnInit {
         }
     }
 
-    handleClose() {
+    handleClose(result?, fromService?) {
+        this.elRef.nativeElement.style.display = 'none';
+        const alertDiv = <HTMLElement>this.elRef.nativeElement.querySelector('.fd-alert');
+        alertDiv.style.display = 'none';
         this.close.emit(this.id);
+        if (!fromService) {
+            this.alertService.popAlert();
+        }
+    }
+
+    open() {
+        // check to make sure this alert is not already opened
+        if (this.elRef.nativeElement.style.display !== 'block') {
+            const top = this.getTop();
+            this.elRef.nativeElement.style.display = 'block';
+            const alertDiv = <HTMLElement>this.elRef.nativeElement.querySelector('.fd-alert');
```

Comment:
Are we OK with using querySelector? I always try to avoid it in favour of ViewChild/ViewChildren if I can.

Reply:
Good question - in this specific example, we're querying for the div that contains the classname `'.fd-alert'`, which is actually a child element of the `<fd-alert>` component.  ViewChild would be useful if we were trying to select a child `<fd-alert>`component, but, `this` is the alert component itself, so ViewChild wouldn't work here